newsroom/monitoring/views.py

```diff
@@ -83,103 +79,120 @@ def process_form_request(updates, request_updates, form):
         updates["keywords"] = request_updates["keywords"]
 
 
-def get_monitoring_for_company(user):
-    try:
-        company = user["company"] if user and user.get("company") else None
-        return list(query_resource("monitoring", lookup={"company": company}))
-    except KeyError:
-        return []
+async def get_monitoring_for_company(user: UserResourceModel | None):
+    company = user.company if user else None
+    cursor = await MonitoringProfileService().search({"company": company})
+    return await cursor.to_list_raw()
+
 
+class MonitoringIdUrlArg(BaseModel):
+    id: ObjectId
 
-@blueprint.route("/monitoring/<id>/users", methods=["POST"])
-@account_manager_only
-async def update_users(id):
+
+@monitoring_endpoints.endpoint("/monitoring/<id>/users", methods=["POST"], auth=[auth_rules.account_manager_only])
```

Comment:
```suggestion
@monitoring_endpoints.endpoint("/monitoring/<TYPE:id>/users", methods=["POST"], auth=[auth_rules.account_manager_only])
```
Should we have the type here?